tests/utils/test_number_conversion.py

```diff
@@ -4,22 +4,29 @@
 
 
 @pytest.mark.parametrize(
-    "num_ranks,global_num_samples,expected",
-    [(2, 10, 5), (2, 11, 5)],
+    "num_ranks,global_num_samples,local_micro_batch_size,expected",
+    [(2, 100, 10, 5), (2, 110, 10, 5)],
```

Comment:
```suggestion
    [(2, 100, 10, 5), (2, 110, 10, 5), (4, 100, 10, 2), (4, 100, 5, 5)]
```
might be good to add more examples, with `num_ranks != 2` and `local_micro_batch_size != 10`